apps/prairielearn/src/schemas/infoAssessment.ts

```diff
@@ -186,71 +209,97 @@ export const ZoneQuestionJsonSchema = QuestionPointsJsonSchema.extend({
-    )
-    .optional(),
-  questions: z.array(ZoneQuestionJsonSchema).min(1).describe('Array of questions in the zone.'),
-  advanceScorePerc: AdvanceScorePercJsonSchema.optional(),
-  gradeRateMinutes: z
-    .number()
-    .gte(0)
-    .describe(
-      'Minimum amount of time (in minutes) between graded submissions to the same question.',
-    )
-    .optional(),
-  canSubmit: uniqueArray(z.string())
-    .describe(
-      'A list of group role names that can submit questions in this zone. Only applicable for group assessments.',
-    )
-    .optional(),
-  canView: uniqueArray(z.string())
-    .describe(
-      'A list of group role names that can view questions in this zone. Only applicable for group assessments.',
-    )
-    .optional(),
-});
+export const ZoneAssessmentJsonSchema = z
+  .object({
+    title: z
+      .string()
+      .describe(
+        'Zone title, displayed to the students at the top of the question list for the zone.',
+      )
+      .optional(),
+    comment: CommentJsonSchema.optional(),
+    // Do we need to allow for additional keys?
+    comments: CommentJsonSchema.optional().describe('DEPRECATED -- do not use.').meta({
+      deprecated: true,
+    }),
+    maxPoints: z
+      .number()
+      .describe(
+        'Only this many of the points that are awarded for answering questions in this zone will count toward the total points.',
+      )
+      .optional(),
+    numberChoose: z
+      .number()
+      .int()
+      .gte(0)
+      .describe('Number of questions to choose from this zone.')
+      .optional(),
+    bestQuestions: z
+      .number()
+      .int()
+      .gte(0)
+      .describe(
+        'Only this many of the questions in this zone, with the highest number of awarded points, will count toward the total points.',
+      )
+      .optional(),
+    questions: z.array(ZoneQuestionJsonSchema).min(1).describe('Array of questions in the zone.'),
+    advanceScorePerc: AdvanceScorePercJsonSchema.optional(),
+    gradeRateMinutes: z
+      .number()
+      .gte(0)
+      .describe(
+        'Minimum amount of time (in minutes) between graded submissions to the same question.',
+      )
+      .optional(),
+    canSubmit: z
+      .array(z.string())
+      .describe(
+        'A list of group role names that can submit questions in this zone. Only applicable for group assessments.',
+      )
+      .optional()
+      .meta({
+        uniqueItems: true,
+      }),
+    canView: z
+      .array(z.string())
+      .describe(
+        'A list of group role names that can view questions in this zone. Only applicable for group assessments.',
+      )
+      .optional()
+      .meta({
+        uniqueItems: true,
+      }),
+  })
+  .meta({
+    id: 'ZoneAssessmentJsonSchema',
+  });
 
 export const AssessmentJsonSchema = z
-  .object({
+  .strictObject({
     comment: CommentJsonSchema.optional(),
     uuid: z
```

Comment:
Thoughts on `z.uuid()` or `z.guid()`?

Reply:
https://www.postgresql.org/docs/current/datatype-uuid.html
> The data type uuid stores Universally Unique Identifiers (UUID) as defined by [RFC 4122](https://datatracker.ietf.org/doc/html/rfc4122), ISO/IEC 9834-8:2005, and related standards.

https://zod.dev/v4/changelog#zstring-updates

> The z.uuid() now validates UUIDs more strictly against the RFC 9562/4122 specification; specifically, the variant bits must be 10 per the spec. For a more permissive "UUID-like" validator, use z.guid().

Reply:
I would prefer to save that for a separate PR in the interest of moving this forward.

Reply:
Peter is now tracking this in https://github.com/PrairieLearn/PrairieLearn/issues/12591.